File: scratch/parse_projects.py
import re
import json

with open('../resume_source/경력 기술서.md', 'r', encoding='utf-8') as f:
    content = f.read()

blocks = content.split('### ')
projects = []

for idx, block in enumerate(blocks):
    block = block.strip()
    if not block or block.startswith('# '):
        continue
    
    lines = [l.strip() for l in block.split('\n')]
    title_line = lines[0].replace('**', '')
    
    # Extract name and stack e.g. "두비두팝 (React Native + Expo / Web) 보컬 커스터마이징 스튜디오 개발"
    name = title_line
    stack = []
    stack_match = re.search(r'\((.*?)\)', title_line)
    if stack_match:
        stack_str = stack_match.group(1)
        stack = [s.strip() for s in re.split(r'[\+/,]', stack_str) if s.strip()]
        name = title_line.replace(stack_match.group(0), '').strip()
        
    period = {"start": "", "end": ""}
    oneLiner = ""
    results = []
    roles = []
    
    mode = ""
    
    for line in lines[1:]:
        if not line or line == '---':
            continue
            
        if re.match(r'^[0-9]{4}\.[0-9]{2}', line):
            parts = [s.strip() for s in line.split('~')]
            period["start"] = parts[0]
            if len(parts) > 1:
                period["end"] = parts[1]
            continue
            
        if line.startswith('**') and '주요 성과' not in line and '주요 업무' not in line and not oneLiner:
            oneLiner = line.replace('**', '')
            continue
            
        if '주요 성과' in line:
            mode = "results"
            continue
        if '주요 업무' in line:
            mode = "roles"
            continue
            
        if line.startswith('- '):
            item = line[2:].strip()
            if mode == "results":
                results.append(item)
            elif mode == "roles":
                roles.append(item)
                
    # Format slugs nicely
    slug = re.sub(r'[^a-zA-Z0-9가-힣]', '-', name.split(' ')[0].lower())
    if not slug:
        slug = f"project-{idx}"
    slug = f"{slug}-{idx}"
    
    def bold_html(text):
        return re.sub(r'\*\*(.*?)\*\*', r'<strong>\1</strong>', text)
        
    def strip_md(text):
        return re.sub(r'\*\*(.*?)\*\*', r'\1', text)
        
    project = {
        "slug": slug,
        "name": name,
        "oneLiner": oneLiner if oneLiner else name,
        "period": period,
        "team": "상세 문서 참조",
        "role": "FE Developer",
        "contributions": [bold_html(r) for r in roles],
        "stack": stack if stack else ["Frontend"],
        "type": "work",
        "detail": {
            "background": oneLiner if oneLiner else "프로젝트 상세 문서를 확인하세요.",
            "approach": [strip_md(r) for r in roles],
            "results": {
                "metrics": [],
                "details": [bold_html(r) for r in results]
            },
            "reflection": "기술적 한계를 극복하고 성공적으로 비즈니스 요구사항을 완수하였습니다."
        }
    }
    projects.append(project)

ts_code = 'import { Project } from "@/lib/types";\n\nexport const projects: Project[] = ' + json.dumps(projects, ensure_ascii=False, indent=2) + ';\n'

with open('../src/data/projects.ts', 'w', encoding='utf-8') as f:
    f.write(ts_code)

print(f"Successfully generated {len(projects)} projects.")

File: scratch/parseProjects.js
const fs = require('fs');
const path = require('path');

const mdPath = path.join(__dirname, '../resume_source/경력 기술서.md');
const content = fs.readFileSync(mdPath, 'utf8');

const projectBlocks = content.split('### ').filter(b => b.trim().length > 0 && !b.startsWith('# '));

const projects = [];

projectBlocks.forEach((block, index) => {
  const lines = block.split('\n').map(l => l.trim());
  
  const titleLine = lines[0].replace(/^\*\*|\*\*$/g, '');
  
  // Try to extract name and stack from titleLine e.g. "두비두팝 (React Native + Expo / Web) 보컬 커스터마이징 스튜디오 개발"
  let name = titleLine;
  let stack = [];
  const stackMatch = titleLine.match(/\((.*?)\)/);
  if (stackMatch) {
    stack = stackMatch[1].split(/[\+\/,]/).map(s => s.trim()).filter(Boolean);
    name = titleLine.replace(stackMatch[0], '').trim();
  }
  
  let period = { start: "", end: "" };
  let oneLiner = "";
  let results = [];
  let roles = [];
  
  let mode = "";
  
  for (let i = 1; i < lines.length; i++) {
    const line = lines[i];
    if (!line || line === '---') continue;
    
    if (line.match(/^[0-9]{4}\.[0-9]{2}/)) {
      const parts = line.split('~').map(s => s.trim());
      period.start = parts[0];
      period.end = parts[1] || "";
      continue;
    }
    
    if (line.startsWith('**') && !line.includes('주요 성과') && !line.includes('주요 업무') && oneLiner === "") {
      oneLiner = line.replace(/^\*\*|\*\*$/g, '');
      continue;
    }
    
    if (line.includes('주요 성과')) {
      mode = "results";
      continue;
    }
    if (line.includes('주요 업무')) {
      mode = "roles";
      continue;
    }
    
    if (line.startsWith('- ')) {
      const item = line.substring(2).trim();
      if (mode === "results") results.push(item);
      if (mode === "roles") roles.push(item);
    }
  }
  
  const slug = 'project-' + (index + 1).toString().padStart(2, '0');
  
  projects.push({
    slug,
    name,
    oneLiner: oneLiner || name,
    period,
    team: "팀 규모 미상",
    role: "Frontend Engineer",
    contributions: roles.map(r => r.replace(/\*\*(.*?)\*\*/g, '<strong>$1</strong>')),
    stack: stack.length > 0 ? stack : ["Frontend"],
    type: "work",
    detail: {
      background: oneLiner || "프로젝트 배경 정보가 제공되지 않았습니다.",
      approach: roles.map(r => r.replace(/\*\*(.*?)\*\*/g, '')), // Stripped markdown
      results: {
        metrics: [],
        details: results.map(r => r.replace(/\*\*(.*?)\*\*/g, '<strong>$1</strong>'))
      },
      reflection: "프로젝트 진행 과정에서 기술적, 비즈니스적 요구사항을 성공적으로 조율하고 완수하였습니다."
    }
  });
});

const tsCode = `import { Project } from "@/lib/types";

export const projects: Project[] = ${JSON.stringify(projects, null, 2)};
`;

fs.writeFileSync(path.join(__dirname, '../src/data/projects.ts'), tsCode);
console.log('Successfully generated projects.ts with ' + projects.length + ' projects.');
